src/legacy/core_plugins/telemetry/server/get_telemetry_opt_in.ts

```diff
@@ -43,4 +55,60 @@ export async function getTelemetryOptIn(request: any) {
 
     throw error;
   }
+
+  const { attributes } = savedObject;
+
+  // return `enabled` attribute, unless it's false ...
+  if (attributes.enabled !== false) return attributes.enabled;
+
+  // Additional check if they've already opted out (enabled: false):
+  // - if the Kibana version has changed by at least a minor version,
+  //   set them back to the null state to get prompted again.
+
+  const lastKibanaVersion = attributes.lastVersionChecked;
+
+  // if version hasn't changed, just return enabled value
+  if (lastKibanaVersion === currentKibanaVersion) return attributes.enabled;
+
+  const lastSemver = parseSemver(lastKibanaVersion);
+  const currentSemver = parseSemver(currentKibanaVersion);
+
+  // if both versions aren't valid, just return the current enabled value
+  if (lastSemver == null && currentSemver == null) {
+    return attributes.enabled;
+  }
+
+  // if one version is valid and one isn't, return null to force re-prompt
+  if (lastSemver == null || currentSemver == null) {
+    return null;
+  }
+
+  // actual major/minor version comparison, for cases when to return null
+  if (currentSemver.major > lastSemver.major) return null;
+  if (currentSemver.major === lastSemver.major) {
+    if (currentSemver.minor > lastSemver.minor) return null;
+  }
+
+  // current version X.Y is not greater than last version X.Y, return enabled
+  return attributes.enabled;
+}
+
+function parseSemver(version: string | null): semver.SemVer | null {
+  if (version == null) return null;
+
+  // semver functions both return nulls AND throw exceptions: "it depends!"
+  let validated: string | null;
+  try {
+    validated = semver.valid(version);
+  } catch (err) {
+    return null;
+  }
+
+  if (validated == null) return null;
+
+  try {
+    return semver.parse(version);
```

Comment:
can we put this under the same `try/catch`. we can initialize `validated = null` and `return validated` once at the end of the function.

Reply:
I ended up removing the validate before the parse, now just do the parse, since the parse is essentially doing a validate anyway.  Simplified the function quite a bit, now it's just a cover that wraps the potential exception being thrown, returning a null.